### PR TITLE
Show Search Console and PSI widgets on shared dashboard even when Analytics is not connected

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #7116

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/index.js [hunk 1]
```diff
@@ -61,7 +61,10 @@ function DashboardAllTrafficWidget( props ) {
 			return true;
 		}
 
-		return select( CORE_USER ).canViewSharedModule( 'analytics' );
+		return (
+			select( CORE_USER ).canViewSharedModule( 'analytics' ) ||
+			select( CORE_USER ).canViewSharedModule( 'analytics-4' )
+		);
```
Comment (nfmohit) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/index.js:67]: I understand why `canViewSharedAnalytics` was updated for  `DashboardViewIndicator`, great catch there, thank you!

To confirm, is it necessary to check if `analytics-4` is a viewable shared module within the UA dashboard view specific dashboard widget components, provided that `canViewSharedAnalytics` here is only used as a prerequisite for calling `modules/analytics` selectors?

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Overview/index.js [hunk 2]
```diff
@@ -86,7 +86,10 @@ const Overview = ( {
 			return true;
 		}
 
-		return select( CORE_USER ).canViewSharedModule( 'analytics' );
+		return (
+			select( CORE_USER ).canViewSharedModule( 'analytics' ) ||
+			select( CORE_USER ).canViewSharedModule( 'analytics-4' )
+		);
```
Comment (nfmohit) [inline assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Overview/index.js:92]: Same request for confirmation/question as above.

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.js [hunk 3]
```diff
@@ -80,7 +80,10 @@ const SearchFunnelWidget = ( { Widget, WidgetReportError } ) => {
 			return true;
 		}
 
-		return select( CORE_USER ).canViewSharedModule( 'analytics' );
+		return (
+			select( CORE_USER ).canViewSharedModule( 'analytics' ) ||
+			select( CORE_USER ).canViewSharedModule( 'analytics-4' )
+		);
```
Comment (nfmohit) [inline assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.js:86]: Same request for confirmation/question as above.

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for such structured work on this, @tofumatt & @aaemnnosttv!

I have left a small question below, but otherwise, this is looking fantastic!

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/index.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/index.js:67]: That's a great point, these settings are mirrored so no. I'll remove them 👍🏻

**Review by nfmohit — APPROVED** [review]
Looks perfect to me now, thank you @tofumatt!

**Review by aaemnnosttv — APPROVED** [review]
Great!

**Pull request merged**